src/main/java/uk/gov/ons/ctp/response/action/service/ActionProcessingService.java

```diff
@@ -35,66 +33,119 @@
   public static final String CREATED = "CREATED";
   public static final String ENABLED = "ENABLED";
   public static final String NOTIFY = "Notify";
+  public static final String PRINTER = "Printer";
   public static final String PENDING = "PENDING";
 
-  @Autowired private ActionRepository actionRepo;
-
-  @Autowired private ActionInstructionPublisher actionInstructionPublisher;
+  @Autowired private ActionCaseRepository actionCaseRepo;
 
   @Autowired private NotifyService notifyService;
 
-  @Autowired
-  private StateTransitionManager<ActionDTO.ActionState, ActionDTO.ActionEvent>
-      actionSvcStateTransitionManager;
+  @Autowired private NotificationFileCreator notificationFileCreator;
+
+  @Autowired private ActionStateService actionStateService;
 
-  private ActionRequestDecorator[] decorators;
+  @Autowired private ActionRequestContextFactory decoratorContextFactory;
 
-  public abstract ActionRequestContextFactory getActionRequestDecoratorContextFactory();
+  private static final ActionRequestDecorator[] DECORATORS = {
+    new ActionAndActionPlan(),
+    new CaseAndCaseEvent(),
+    new CollectionExerciseAndSurvey(),
+    new PartyAndContact(),
+    new SampleUnitRef()
+  };
 
-  public ActionProcessingService(ActionRequestDecorator[] decorators) {
-    this.decorators = decorators;
+  public void processActions(ActionType actionType, List<Action> allActions) {
+    if (actionType == null) {
+      throw new IllegalStateException(ACTION_TYPE_NOT_DEFINED);
```

Comment:
Would NullPointerException be more appropriate here?  I'm fine with this to remain as is though as what's here is correct too

Reply:
I wanted to maintain the current functionality and the old validate method threw illegal state exception if the action type was null. To be fair, it will never happen.

private boolean validate(final Action action) {
    final ActionType actionType = action.getActionType();
    if (!valid(actionType)) {
      log.with("action", action).error("ActionType is not defined for action");
      throw new IllegalStateException(ACTION_TYPE_NOT_DEFINED);
    } 
  }

Reply:
ah fair play.  Considering the size of the changes it might be worth leaving it as is for now?  Your choice